### PR TITLE
Memory allocation bugs

### PR DESCRIPTION
**Pull request by douglascaetano**
Hi!

As asked on PR #1137, I detached some commits and made this new PR. The following commits fix some problems with memory usage and allocation. Please refer to each commits message for more details.

Thanks!

### CODE REVIEW

File: text2pcap.c [hunk 1]
```diff
@@ -50,57 +50,60 @@ const char *janus_text2pcap_packet_string(janus_text2pcap_packet type) {
 }
 
 janus_text2pcap *janus_text2pcap_create(const char *dir, const char *filename, int truncate) {
+	janus_text2pcap *tp;
+	char newname[1024];
+	char *fname;
+	FILE *f;
```
Comment (lminiero) [inline text2pcap.c]: All of the above pointers should be initialized to NULL.

**Comment by lminiero** [comment]
@douglascaetano now that we merged #1137 I'll wait for the tweaks to this patch (I see there now is a conflict though?) to merge this too.

File: text2pcap.c (hunk 1 above)
Comment (douglascaetano) [reply text2pcap.c]: Actually, there is no need for it. If you follow the code, you'll see:

- `*tp` will be initialized with the allocation right before being used for the first time;
- `*f` will be initialized in both branches of the first `if`, also before the first use; and
- `*fname` will be initialized whenever `*f` is not set to `NULL` in the previous assignments, and when `*f` is in fact `NULL`, the code that uses `*fname` will never be reached.

But I admit the code is a bit confusing, so a refactoring is on the way. :)

**Comment by douglascaetano** [comment]
@lminiero, do you mind if I do a rebase and a forced push so git history is kept cleaner?

**Comment by lminiero** [comment]
> @lminiero, do you mind if I do a rebase and a forced push so git history is kept cleaner?

No problem at all, thx!

**Comment by douglascaetano** [comment]
Done! I took off some complexity on that function. Hope it's easier to understand now! :grinning:

**Comment by douglascaetano** [comment]
By the way, why do you try `fopen()` twice? AFAIK, if an error occurs with "a" mode, the same would occur for "w" mode, wouldn't?

**Comment by lminiero** [comment]
> By the way, why do you try fopen() twice? AFAIK, if an error occurs with "a" mode, the same would occur for "w" mode, wouldn't?

Not sure, but the plan was to try appending first, and if that failed create the file from scratch. Not sure if trying to append to a non-existing file creates a new file nevertheless, in which case the second fopen is indeed useless.

**Comment by douglascaetano** [comment]
I see. I asked in case you had some trouble before. Indeed, according to the [`fopen()` man page](http://man7.org/linux/man-pages/man3/fopen.3.html), the file is created if it does not exist in either `w` or `a` modes, so it is safe in this regard. If there are no objections, can I add a commit for this?

**Comment by lminiero** [comment]
Sure, makes sense to add it as part of the fixes, thx!

**Comment by douglascaetano** [comment]
Done! :blush:

**Comment by lminiero** [comment]
Thx, merging then!

**Pull request merged**